watchtower/wtmock/peer.go

```diff
@@ -175,6 +175,45 @@ func (p *MockPeer) SetDeadline(t time.Time) error {
 	panic("not implemented")
 }
 
+// WriteMessage mocks sending the message.
+func (p *MockPeer) WriteMessage(msg []byte) error {
+	// Since we may be using a write pool, we copy the message
+	// over before sending it through.
+	msgLen := len(msg)
+	msgCopy := make([]byte, msgLen)
+	if msgLen != copy(msgCopy, msg) {
+		return fmt.Errorf("failed to copy message: %v", msg)
+	}
+
+	select {
+	case p.OutgoingMsgs <- msgCopy:
+	case <-time.After(5 * time.Second):
+		return fmt.Errorf("timeout sending message: %v", msg)
+	}
+
+	return nil
+}
+
+// Flush mocks flushing the message down the wire.
+func (p *MockPeer) Flush() (int, error) {
+	return 0, nil
+}
+
+// ReadNextHeader mocks reading the next header.
+func (p *MockPeer) ReadNextHeader() (uint32, error) {
+	return 0, nil
+}
+
+// ReadNextBody mocks reading the next body.
+func (p *MockPeer) ReadNextBody(_ []byte) ([]byte, error) {
+	select {
+	case msg := <-p.IncomingMsgs:
+		return msg, nil
```

Comment:
nit: new line before the next `case` branch

Reply:
done